frontend/simple/js/backend/hapi.js

```diff
@@ -73,29 +73,55 @@ export class HapiBackend extends Backend {
   subscriptions () {
     return Object.keys(store.state.contracts)
   }
-  async subscribe (contractId: string) {
+  async subscribeAndSyncIdentity (username: string, ephemeral: bool) {
+    let identity = await namespace.lookup(username)
+    await this.subscribeAndSync(identity, ephemeral)
+  }
+  async unsubscribeIdentity (username: string, ephemeral: bool) {
+    let identity = await namespace.lookup(username)
+    await this.unsubscribe(identity, ephemeral)
+  }
```

Comment:
So if the purpose of this function is to keep DRY... as I'm guessing it is (which is great!) ... maybe we need to find a different place to put it?

I'm only just starting with the review so am thinking out loud here.